packages/migrations/src/migrations/migrations.ts

```diff
@@ -53,8 +53,14 @@ export function getMigrationsToExecute(
   return migrationFiles.filter((m) => !executedMigrationTimestamps.has(m.timestamp));
 }
 
+let didMigration = false;
+
 export async function initWithLock(directories: string[], project: string) {
+  // For Vite HMR mode
+  if (didMigration && (import.meta as any).env?.DEV) return;
+
   logger.verbose('Starting DB schema migration');
+  didMigration = true;
```

Comment:
Why is this flag needed? Migration execution should always be idempotent.

Reply:
See https://github.com/PrairieLearn/PrairieLearn/pull/11914#issuecomment-2851706325